cmd/k8sagent/unit/manifolds.go

```diff
@@ -0,0 +1,330 @@
+
+	// PreviousAgentVersion passes through the version the unit
+	// agent was running before the current restart.
+	PreviousAgentVersion version.Number
+
+	// UpgradeStepsLock is passed to the upgrade steps gate to
+	// coordinate workers that shouldn't do anything until the
+	// upgrade-steps worker is done.
+	UpgradeStepsLock gate.Lock
+
+	// PreUpgradeSteps is a function that is used by the upgradesteps
+	// worker to ensure that conditions are OK for an upgrade to
+	// proceed.
+	PreUpgradeSteps func(*state.StatePool, coreagent.Config, bool, bool, bool) error
+
+	// MachineLock is a central source for acquiring the machine lock.
+	// This is used by a number of workers to ensure serialisation of actions
+	// across the machine.
+	MachineLock machinelock.Lock
+
+	// Clock contains the clock that will be made available to manifolds.
+	Clock clock.Clock
+}
+
+// Manifolds returns a set of co-configured manifolds covering the various
+// responsibilities of a k8s agent unit command. It also accepts the logSource
+// argument because we haven't figured out how to thread all the logging bits
+// through a dependency engine yet.
+//
+// Thou Shalt Not Use String Literals In This Function. Or Else.
+func Manifolds(config manifoldsConfig) dependency.Manifolds {
+	return dependency.Manifolds{
+		// The agent manifold references the enclosing agent, and is the
+		// foundation stone on which most other manifolds ultimately depend.
+		agentName: agent.Manifold(config.Agent),
+
+		// The api-config-watcher manifold monitors the API server
+		// addresses in the agent config and bounces when they
+		// change. It's required as part of model migrations.
+		apiConfigWatcherName: apiconfigwatcher.Manifold(apiconfigwatcher.ManifoldConfig{
+			AgentName:          agentName,
+			AgentConfigChanged: config.AgentConfigChanged,
+			Logger:             loggo.GetLogger("juju.worker.apiconfigwatcher"),
+		}),
+
+		apiCallerName: apicaller.Manifold(apicaller.ManifoldConfig{
+			AgentName:            agentName,
+			APIOpen:              api.Open,
+			APIConfigWatcherName: apiConfigWatcherName,
+			NewConnection:        apicaller.OnlyConnect,
+			Logger:               loggo.GetLogger("juju.worker.apicaller"),
+		}),
+
+		clockName: clockManifold(config.Clock),
+
+		// The log sender is a leaf worker that sends log messages to some
+		// API server, when configured so to do. We should only need one of
+		// these in a consolidated agent.
+		logSenderName: logsender.Manifold(logsender.ManifoldConfig{
+			APICallerName: apiCallerName,
+			LogSource:     config.LogSource,
+		}),
+
+		// The upgrade steps gate is used to coordinate workers which
+		// shouldn't do anything until the upgrade-steps worker has
+		// finished running any required upgrade steps. The flag of
+		// similar name is used to implement the isFullyUpgraded func
+		// that keeps upgrade concerns out of unrelated manifolds.
+		upgradeStepsGateName: gate.ManifoldEx(config.UpgradeStepsLock),
```

Comment:
We can remove the upgrade steps, gates, flags, and checks from here.

Perhaps leaving a comment that this agent doesn't any upgrade steps checks because it will just be restarted when upgrades happen.